sdk/log/DESIGN.md

```diff
@@ -0,0 +1,200 @@
+# Logs SDK
+
+## Abstract
+
+`go.opentelemetry.io/otel/sdk/log` provides Logs SDK compliant with the
+[specification](https://opentelemetry.io/docs/specs/otel/logs/sdk/).
+
+The main and recommended use case is to configure the SDK to use an OTLP
+exporter with a batch processor.[^1] Therefore, the design aims to be
+high-performant in this scenario.
+
+The prototype was created in
+[#4955](https://github.com/open-telemetry/opentelemetry-go/pull/4955).
+
+## Modules structure
+
+The SDK is published as a single `go.opentelemetry.io/otel/sdk/log` Go module.
+
+The exporters are going to be published as following Go modules:
+
+- `go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc`
+- `go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp`
+- `go.opentelemetry.io/otel/exporters/stdout/stdoutlog`
+
+## LoggerProvider
+
+The [LoggerProvider](https://opentelemetry.io/docs/specs/otel/logs/sdk/#loggerprovider)
+is defined as follows:
+
+```go
+type LoggerProvider struct {
+	embedded.LoggerProvider
+}
+
+func NewLoggerProvider(...Option) *LoggerProvider
+
+func (*LoggerProvider) Logger(name string, options ...log.LoggerOption) log.Logger
+
+type Option interface { /* ... */ }
+
+func WithResource(*resource.Resource) Option
+```
+
+## LogRecord limits
+
+The [LogRecord limits](https://opentelemetry.io/docs/specs/otel/logs/sdk/#logrecord-limits)
+are defined as follows:
+
+```go
+func WithLimits(Limits) Option
```

Comment:
I plan to change it to `WithAttributeValueLengthLimit` and `WithAttributeCountLimit` instead.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4954/commits/2e54ea5f55b2ab10316810d2b50b3d5946d0ecba